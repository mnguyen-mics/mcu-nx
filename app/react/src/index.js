import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
