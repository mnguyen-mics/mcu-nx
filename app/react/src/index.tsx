/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
