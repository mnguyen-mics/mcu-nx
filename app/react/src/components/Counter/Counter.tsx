import * as React from 'react';
import McsIcon, { McsIconType } from '../McsIcon';
import { Spin } from 'antd';
import { FormattedNumber } from 'react-intl';

export interface LoadingCounterValue {
  value?: number;
  loading?: boolean;
}

export interface CounterProps extends LoadingCounterValue {
  iconType: McsIconType;
  iconStyle?: React.CSSProperties;
  unit?: string;
  title: React.ReactNode | string;
}

export default class Counter extends React.Component<CounterProps> {
  render() {
    const { iconType, iconStyle, unit, title, value, loading } = this.props;

    return (
      <div className="counter" >
        <McsIcon type={iconType} style={iconStyle ? iconStyle : {}}/>
        <div className="counter_title">{title}</div>
        <div className="number">
          {loading ? (
            <Spin />
          ) : (value !== undefined && value !== null) ? (
            <span><FormattedNumber value={value} />{unit || ''}</span>
          ) : (
            '--'
          )}
        </div>
      </div>
    );
  }
}
