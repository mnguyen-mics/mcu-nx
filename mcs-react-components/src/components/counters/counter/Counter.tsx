import * as React from 'react';
import McsIcon, { McsIconType } from '../../mcs-icon';
import { UpOutlined, DownOutlined } from '@ant-design/icons';
import { Spin, Statistic } from 'antd';
import { FormattedNumber } from 'react-intl';

export interface LoadingCounterValue {
  value?: number;
  loading?: boolean;
}

export interface Trend {
  value: number;
  type: 'up' | 'down';
}

export interface CounterProps extends LoadingCounterValue {
  iconType: McsIconType;
  iconStyle?: React.CSSProperties;
  title: React.ReactNode | string;
  unit?: string;
  trend?: Trend;
}

export default class Counter extends React.Component<CounterProps> {
  render() {
    const { iconType, iconStyle, title, value, loading, unit, trend } = this.props;

    return (
      <div className="mcs-counter" >
        <McsIcon type={iconType} styleIcon={iconStyle ? iconStyle : {}} />
        <div className="title">{title}</div>
        <div className="number">
          {loading ? (
            <Spin />
          ) : (value !== undefined && value !== null) ? (
            <React.Fragment>
              <FormattedNumber value={value} />{unit ? ' ' + unit : ''}
              {trend ? <Statistic
                title=""
                value={trend.value}
                precision={1}
                valueStyle={{ color: trend.type === 'up' ? '#4ea500' : '#ed2333' }}
                prefix={trend.type === 'up' ? <UpOutlined /> : <DownOutlined />}
                suffix="%"
              /> : null}
            </React.Fragment>
          ) : (
                '--'
              )}
        </div>
      </div>
    );
  }
}
