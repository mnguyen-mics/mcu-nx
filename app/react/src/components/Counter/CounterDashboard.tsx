import * as React from 'react';
import Counter, { CounterProps } from './Counter';
import { Row, Col } from 'antd/lib/grid';

export interface Props {
  counters: CounterProps[];
  invertedColor?: boolean;
}

export default class CounterDashboard extends React.Component<Props> {
  render() {
    const { counters, invertedColor } = this.props;

    if (counters.length === 0) return null;

    const colStyle: React.CSSProperties = {
      width: `${(100 / counters.length).toFixed(2)}%`,
      float: 'left',
      flex: '0 0 auto',
    };
    return (
      <div className={`counter-dashboard ${invertedColor ? 'inverted' : ''}`}>
        <Row>
          {counters.map((counter, index) => {
            return (
              <Col
                key={index}
                style={colStyle}
                className={index !== counters.length - 1 ? 'border-right' : ''}
              >
                <Counter {...counter} />
              </Col>
            );
          })}
        </Row>
      </div>
    );
  }
}
