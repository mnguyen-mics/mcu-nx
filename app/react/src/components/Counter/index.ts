import CounterDashboard from './CounterDashboard';
import Counter from './Counter';

export {
  CounterDashboard,
  Counter,
}