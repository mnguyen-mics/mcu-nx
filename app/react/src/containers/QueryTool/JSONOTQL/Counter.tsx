import * as React from 'react';
import { Icon } from 'antd';
import { formatMetric } from '../../../utils/MetricHelper';
import { McsIcon } from '../../../components';

export interface CounterProps {
  name: string;
  value?: number;
  loading?: boolean;
  error?: boolean;
  stale?: boolean;
  onRefresh: () => void;
  width: string;
  viewValue?: boolean;
}

interface State {
  hover: boolean;
}

export default class Counter extends React.Component<CounterProps, State> {
  constructor(props: CounterProps) {
    super(props);
    this.state = { hover: false };
  }

  render() {
    const {
      name,
      value,
      loading,
      error,
      stale,
      onRefresh,
      width,
      viewValue,
    } = this.props;

    const onHover = (type: 'enter' | 'leave') => () =>
      this.setState({ hover: type === 'enter' ? true : false });

    const restOfCounter =
      viewValue !== false ? (
        <React.Fragment>
          <div className="view-value">
            {loading ? (
              <i
                className="mcs-table-cell-loading"
                style={{ maxWidth: '100%' }}
              />
            ) : (
              formatMetric(value, '0,0')
            )}
          </div>
          {loading && <div className={'refresh-overlay'} onClick={onRefresh} />}
          {loading && (
            <div className="refresh-text">
              <Icon type="loading" />
            </div>
          )}
          {(stale || this.state.hover) && !loading && (
            <div className={'refresh-overlay'} onClick={onRefresh} />
          )}
          {(stale || this.state.hover) && !loading && (
            <div className="refresh-text" onClick={onRefresh}>
              <McsIcon type="refresh" />
            </div>
          )}
          {error && !loading && (
            <div className={'refresh-overlay error'} onClick={onRefresh} />
          )}
          {error && !loading && (
            <div className="refresh-text error" onClick={onRefresh}>
              <McsIcon type="refresh" />
            </div>
          )}
        </React.Fragment>
      ) : null;

    return (
      <div
        className="m-t-20"
        style={{
          float: 'left',
          width: width,
          padding: '0 8px',
          position: 'relative',
        }}
        onMouseEnter={onHover('enter')}
        onMouseLeave={onHover('leave')}
      >
        <div className="mcs-card no-margin-bottom result-view">
          <div className="view-name">{name}</div>
          {restOfCounter}
        </div>
      </div>
    );
  }
}
