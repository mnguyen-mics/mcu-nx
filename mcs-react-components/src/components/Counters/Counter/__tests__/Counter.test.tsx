import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Counter, { CounterProps } from '../Counter';
import { IntlProvider } from 'react-intl';

it('renders the Counter', () => {
  const props: CounterProps = {
    iconType: 'users',
    title: 'Users',
    value: 424242,
    loading: false,
  };
  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <Counter {...props} />
    </IntlProvider>,
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
