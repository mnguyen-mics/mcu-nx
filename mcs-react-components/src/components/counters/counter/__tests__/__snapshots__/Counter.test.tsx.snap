// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Counter 1`] = `
<div
  className="mcs-counter "
>
  <span
    className="mcs-icon "
  >
    <i
      className="mcs-users"
      style={Object {}}
    />
  </span>
  <div
    className="title"
  >
    Users
  </div>
  <div
    className="number"
  >
    <span>
      424,242
    </span>
     Users
    <div
      className="ant-statistic"
    >
      
      <div
        className="ant-statistic-content"
        style={
          Object {
            "color": "#4ea500",
          }
        }
      >
        <span
          className="ant-statistic-content-prefix"
        >
          <span
            aria-label="up"
            className="anticon anticon-up"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="up"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
              />
            </svg>
          </span>
        </span>
        <span
          className="ant-statistic-content-value"
        >
          <span
            className="ant-statistic-content-value-int"
          >
            
            20
          </span>
          <span
            className="ant-statistic-content-value-decimal"
          >
            .0
          </span>
        </span>
        <span
          className="ant-statistic-content-suffix"
        >
          %
        </span>
      </div>
    </div>
  </div>
</div>
`;
